tests/test_types.py

```diff
@@ -34,3 +38,36 @@ def test_completely_new(self) -> None:
         added, removed = diff(set(), new)
         self.assertEqual(added, new)
         self.assertFalse(removed)
+
+
+class TestDocumentSelector(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self._opened_views = []  # type: List[sublime.View]
+
+    def tearDown(self) -> None:
+        for view in self._opened_views:
+            view.close()
+        self._opened_views.clear()
+
+    def _make_view(self, syntax: str, file_name: str) -> sublime.View:
+        view = sublime.active_window().new_file(0, syntax)
+        self._opened_views.append(view)
+        view.set_scratch(True)
+        self.assertFalse(view.is_loading())
+        view.file_name = MagicMock(return_value=file_name)
+        return view
+
+    def test_language(self) -> None:
+        selector = DocumentSelector([{"language": "txt"}])
+        view = self._make_view("Packages/Text/Plain text.tmLanguage", "foobar.txt")
+        self.assertTrue(selector.matches(view))
+        view = self._make_view("Packages/Python/Python.sublime-syntax", "hello.py")
+        self.assertFalse(selector.matches(view))
+
```

Comment:
Do we support all features from the spec?

```js
	/**
	 * A glob pattern, like `*.{ts,js}`.
	 *
	 * Glob patterns can have the following syntax:
	 * - `*` to match one or more characters in a path segment
	 * - `?` to match on one character in a path segment
	 * - `**` to match any number of path segments, including none
	 * - `{}` to group conditions (e.g. `**​/*.{ts,js}` matches all TypeScript and JavaScript files)
	 * - `[]` to declare a range of characters to match in a path segment (e.g., `example.[0-9]` to match on `example.0`, `example.1`, …)
	 * - `[!...]` to negate a range of characters to match in a path segment (e.g., `example.[!0-9]` to match on `example.a`, `example.b`, but not `example.0`)
	 */
```

Reply:
It all works :) I had to enable the `BRACE` and `GLOBSTAR` flags.